File: services/gateway/gateway-api/src/main/java/com/example/gateway/api/converter/TradeSummaryConverter.java
package com.example.gateway.api.converter;

import com.example.tradelog.api.spec.model.TradeSummaryItem;

import java.math.BigDecimal;
import java.util.function.Function;

public class TradeSummaryConverter {

    public static Function<TradeSummaryItem, TradeSummaryGWModel> toTradeSummaryGWModel = model ->
            TradeSummaryGWModel.builder()
                    .withSymbol(model.getSymbol())
                    .withTotal(BigDecimal.valueOf(model.getTotal()))
                    .withTrades(model.getTrades())
                    .build();
}

File: services/tradelog/tradelog-api/src/main/kotlin/com/example/tradelog/api/rest/converter/TradeSummaryConverter.kt
package com.example.tradelog.api.rest.converter

import com.example.tradelog.api.core.model.TradeSummaryModel
import com.example.tradelog.api.spec.model.TLTradeSummaryItem
import com.example.tradelog.api.spec.model.TLTradeSummaryResponse
import java.util.stream.Collectors


class TradeSummaryConverter {

    companion object {
        fun toTradeSummaryResponse(models: List<TradeSummaryModel>): TLTradeSummaryResponse {

            val items = models.stream()
                    .map { i -> toTradeSummaryItem(i) }
                    .collect(Collectors.toList())

            return TLTradeSummaryResponse.newBuilder()
                    .addAllItems(items)
                    .build()
        }

        private fun toTradeSummaryItem(model: TradeSummaryModel): TLTradeSummaryItem {
            return TLTradeSummaryItem.newBuilder()
                    .setSymbol(model.symbol)
                    .setTotal(model.total.toDouble())
                    .setTrades(model.trades)
                    .build()
        }
    }

}